docs/reference/api/platform/Logger.md

```diff
@@ -0,0 +1,164 @@
+## Logger
+
+Mbed OS provides a set of API's that you can use to output different log level messages to STDIO at runtime. mbed_logger.h declares these functions, which are available only in debug builds.
+API's in logging module are printf-style API's which take module name and format string followed by arguments.
+
+By default all messages till LOG_LEVEL_DEBUG are enabled. If you want to enable Trace and Info level messages, then set `MBED_CONF_MAX_LOG_LEVEL` accordingly.
+
+### Debug Log Levels
+Below are various log levels supported and recommended usage.
+
+#### LOG_LEVEL_ERR_CRITICAL
+In case of critical errors when system recovery is not possible, you should use `MBED_CRIT` API. `MBED_CRIT` is equivalent to `ASSERT`, it will dump previous debug log and application will terminate. 
```

Comment:
"it will dump previous debug log and application will terminate" - can you please clarify on this? what is previous debug log here? Is that the messages captured in log buffer?

Reply:
Yes the messages which are already captured in log buffer, since we push them to circular buffer. We need to dump them before performing mbed_die() operation. 
You want me to clarify more in document? In that case i will have to explain entire design/implementation of logging module

Reply:
You can say exactly as you mentioned - "all the messages which are already captured in log buffer".